File: lua/resistor-color-duo/resistor-color-duo.lua
local resistor_colors = {
  ["black"] = 0,
  ["brown"] = 1,
  ["red"] = 2,
  ["orange"] = 3,
  ["yellow"] = 4,
  ["green"] = 5,
  ["blue"] = 6,
  ["violet"] = 7,
  ["grey"] = 8,
  ["white"] = 9,
}

return {
  value = function(colors)
    local num = ""

    for i = 1, 2 do -- max two items
      num = num .. resistor_colors[colors[i]]
    end

    return tonumber(num)
  end,
}

File: csharp/resistor-color-duo/ResistorColorDuo.cs
using System.Text;

public static class ResistorColorDuo
{
  private enum ResistorColors { Black, Brown, Red, Orange, Yellow, Green, Blue, Violet, Grey, White }

  public static int Value(string[] colors)
  {
    var sb = new StringBuilder();

    foreach (var c in colors.Take(2))
      sb.Append((int)Enum.Parse<ResistorColors>(c, ignoreCase: true));

    return !int.TryParse(sb.ToString(), out int result)
      ? throw new InvalidOperationException()
      : result;
  }
}
